src/views/Student/Student.js

```diff
@@ -0,0 +1,325 @@
+            title: 'Error',
+            message: 'Failed to fetch student',
+            level: 'error'
+          })
+        }
+      }).catch(err => {
+        console.error(err)
+        this.context.addNotification({
+          title: 'Error',
+          message: 'Failed to fetch student',
+          level: 'error'
+        })
+      })
+    })
+  }
+
+  toggleEdit(){
+    if(this.state.editStudent){
+      let numericKeys = ['behavior','workEthic', 'mathBench', 'cogAT', 'dra', 'elaTotal', 'mathTotal', 'behaviorObservation', 'dial4']
+      for (let i = 0; i < numericKeys.length; i++){
+        let key = numericKeys[i]
+        if( typeof this.state.student[key] === 'string' && this.state.student[key]){
+          let tempStudent = this.state.student
+          tempStudent[key] = parseInt(this.state.student[key],10)
+          this.setState({student : tempStudent })
+        }
+      }
+      let checkedScores = numericKeys.slice(2)
+      for (let i = 0; i < checkedScores.length; i++){
+        let key = checkedScores[i]
+        if(this.validateScore(key) === 'error'){
+          this.context.addNotification({
+            title: 'Error',
+            message: `Invalid value entered for ${Utils.studentTranslations[key]}`,
+            level: 'error'
+          })
+          return
+        }
+      }
+      fetch(`${process.env.REACT_APP_SERVER_ADDRESS}/api/students/${this.props.params.studentID}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          student: this.state.student
+        })
+      }).catch(err => {
+        console.error(err)
+        this.context.addNotification({
+          title: 'Error',
+          message: 'Failed to update student info',
+          level: 'error'
+        })
+      })
+    } else {
+      this.unchangedStudent = JSON.parse(JSON.stringify(this.state.student))
+    }
+    this.setState({ editStudent: !this.state.editStudent})
+  }
+
+  discardChanges(){
+    this.setState({ student: this.unchangedStudent})
+    this.setState({ editStudent: false})
+  }
+
+
+  getInfo(){
+    const { student } = this.state
+    return (
+      <ListGroup fill>
+        {
+          Utils.cardKeys.filter(key => key in Utils.studentTranslations).sort(Utils.sortStudentStats).map((key, i) => {
+            return <Col xs={12} md={6}> <ListGroupItem key={i}>{`${Utils.forHumanAttr(key, student[key])}`}</ListGroupItem> </Col>
+          })
+        }
+      </ListGroup>
+    )
+  }
+
+  getEditForm(){
+    const { student } = this.state
+    return (
+      <div>
+        {
+          Utils.cardKeys.filter(key => key in Utils.studentTranslations).sort(Utils.sortStudentStats).map((key, i) => {
+            return <Col xs={12} md={6}> {this.getFormItem(key, student[key])}</Col>
+          })
+        }
+      </div>
+    )
+  }
+
+  validateScore(key){
+    let val = this.state.student[key]
+    let retVal = ''
```

Comment:
Is using `retVal` necessary? I think we could just return 'error' or 'success' immediately in any of the cases